app/api/role_invites.py

```diff
@@ -0,0 +1,117 @@
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+
+from app.api.helpers.utilities import dasherize
+from app.models import db
+from app.models.event import Event
+from app.models.role_invite import RoleInvite
+from app.api.bootstrap import api
+from app.api.helpers.permissions import jwt_required
+from app.api.helpers.db import safe_query
+
+
+class RoleInviteSchema(Schema):
+    """
+    Api Schema for role invite model
+    """
+
+    class Meta:
+        """
+        Meta class for role invite Api Schema
+        """
+        type_ = 'role-invite'
+        self_view = 'v1.role_invite_detail'
+        self_view_kwargs = {'id': '<id>'}
+        inflect = dasherize
+
+    id = fields.Str(dump_only=True)
+    email = fields.Str(required=True)
+    hash = fields.Str(allow_none=True)
+    created_at = fields.DateTime(dump_only=True, timezone=True)
+    role_id = fields.Integer(required=True)
```

Comment:
Ok. Since I am already using events endpoint, how do I go about  this?
1) use this endpoint: `/events/<int:event_id>/role/<role_id>/role-invites` to get both role_id and event_id during post
or
2) first POST the role_invite without role_id via events endpoint, then POST role_id using relationship?(2 way process)
because I will first need to have a `role_invite` before its `role_id`.